scripts/workers/submission_worker.py

```diff
@@ -547,15 +547,34 @@ def main():
     queue_name = os.environ.get('CHALLENGE_QUEUE', 'evalai_submission_queue')
     queue = get_or_create_sqs_queue(queue_name)
 
-    while True:
-        for message in queue.receive_messages():
-            logger.info('Processing message body: {0}'.format(message.body))
-            process_submission_callback(message.body)
-            # Let the queue know that the message is processed
-            message.delete()
-        if killer.kill_now:
-            break
-        time.sleep(0.1)
+    if settings.DEBUG or settings.TEST:
+        while True:
+            for message in queue.receive_messages():
+                logger.info('Processing message body: {0}'.format(message.body))
+                process_submission_callback(message.body)
+                # Let the queue know that the message is processed
+                message.delete()
+            if killer.kill_now:
+                break
+            time.sleep(0.1)
+    else:
+        maximum_concurrent_submissions = challenges.evaluate_concurrent_max_submissions
+        while True:
+            submissions_done_in_challenge_count = Submission.objects.filter(
```

Comment:
Shouldn't this check be after line 569 in this PR? Right now, you will basically be querying the database many times each second to get the number of running submissions which doesn't make sense.